### PR TITLE
Meaningful message returned if schedule name too long

### LINKED ISSUE #353: Need a meaningful error message when schedule names are too long
Currently the deployer returns the following exceptions if your schedule name exceeds the maximum char size for the field.
When creating a container the constraint reports `must be between 0 and 63`. 
In the last version a KubernetesClientException was fired.   In this case reported that the Schedule  was not created but no reason why.   
We need to let the user know that their schedule name exceeded maximum allowed characters.   

This issue is a child of https://github.com/spring-cloud/spring-cloud-dataflow/issues/3682

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #353

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/scheduler/kubernetes/KubernetesScheduler.java [hunk 1]
```diff
@@ -190,4 +210,12 @@ private void setImagePullSecret(ScheduleRequest scheduleRequest, CronJob cronJob
 					.add(localObjectReference);
 		}
 	}
+
+	private boolean isScheduleNameTooLong(String message ) {
```
Comment (chrisjs) [inline src/main/java/org/springframework/cloud/deployer/spi/scheduler/kubernetes/KubernetesScheduler.java]: maybe `validationScheduleNameLength` or similar -- mainly to match something liek the other `validateScheduleName`

File: src/main/java/org/springframework/cloud/deployer/spi/scheduler/kubernetes/KubernetesScheduler.java [hunk 2]
```diff
@@ -190,4 +210,12 @@ private void setImagePullSecret(ScheduleRequest scheduleRequest, CronJob cronJob
...
+		boolean result = false;
+		if(StringUtils.hasText(message) && message.contains("must be no more than") && message.endsWith("characters")) {
```
Comment (chrisjs) [inline src/main/java/org/springframework/cloud/deployer/spi/scheduler/kubernetes/KubernetesScheduler.java]: consider just inlining

**Comment by cppwfs** [comment]
Submitted issue https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/356 so that we can investigate using the Constraints provided by Fabric8 instead of home-rolling our own.

**Comment by chrisjs** [comment]
merged in 034f9c426a8f512d77defcea996f5bc440c31052

**Pull request closed without merging**